### ISSUE TITLE
Failing to build on OS X because of python locale bug (+fix)

### ISSUE THREAD

**Issue by hmijail** [facebook/infer#438]
In OS X 10.11.6, Infer 0.9 failed to build with this stack trace:
```
Traceback (most recent call last):
  File "/Users/mija/Downloads/infer-osx-v0.9.0/infer/bin/../lib/python/infer.py", line 18, in <module>
    from inferlib import analyze, config, utils
  File "/Users/mija/Downloads/infer-osx-v0.9.0/infer/lib/python/inferlib/analyze.py", line 26, in <module>
    from . import config, issues, utils
  File "/Users/mija/Downloads/infer-osx-v0.9.0/infer/lib/python/inferlib/config.py", line 17, in <module>
    locale.setlocale(locale.LC_ALL, '')
  File "/opt/local/Library/Frameworks/Python.framework/Versions/2.7/lib/python2.7/locale.py", line 581, in setlocale
Traceback (most recent call last):
  File "/Users/mija/Downloads/infer-osx-v0.9.0/infer/bin/../lib/python/infer.py", line 18, in <module>
    from inferlib import analyze, config, utils
  File "/Users/mija/Downloads/infer-osx-v0.9.0/infer/lib/python/inferlib/analyze.py", line 26, in <module>
    from . import config, issues, utils
  File "/Users/mija/Downloads/infer-osx-v0.9.0/infer/lib/python/inferlib/config.py", line 17, in <module>
    locale.setlocale(locale.LC_ALL, '')
  File "/opt/local/Library/Frameworks/Python.framework/Versions/2.7/lib/python2.7/locale.py", line 581, in setlocale
    return _setlocale(category, locale)
locale.Error: unsupported locale setting
    return _setlocale(category, locale)
locale.Error: unsupported locale setting
Failed to execute: /Users/mija/Downloads/infer-osx-v0.9.0/infer/bin/../lib/python/infer.py --multicore 4 --out out/ --project_root /Users/mija/Downloads/infer-osx-v0.9.0/infer/models/objc -- /Applications/Xcode.app/Contents/Developer/usr/bin/make -C src
```

This is a known python bug: it doesn't react well to non-Linux locales: https://bugs.python.org/issue18378

These were my locale environment variables: 
```
$ locale
LANG=
LC_COLLATE="C"
LC_CTYPE="C"
LC_MESSAGES="C"
LC_MONETARY="C"
LC_NUMERIC="C"
LC_TIME="C"
LC_ALL=
```

One workaround was to do this: 
`$ export LC_ALL=en_US.UTF-8`

**Comment by jvillard** [comment]
I cannot reproduce your issue with the same locale settings on 10.10.5. That said, a simple fix would be to catch this error when we call `locale.setlocale()` in [config.py](https://github.com/facebook/infer/blob/master/infer/lib/python/inferlib/config.py#L17-L20) and fall back to our default `'ascii'` codeset in that case too.

Would you mind submitting a PR to that effect if that fixes your issue?

**Comment by jvillard** [comment]
I just hit the same bug by running Infer inside a Linux VM inside osx. I'll look into it.

**Comment by jvillard** [comment]
fixed in 49adf5933260836c17bc29b90a6ade50a803a83d
